src/xdocs/config_design.xml

```diff
@@ -1479,34 +1631,44 @@ public class ImmutableClass
         <source>
 &lt;module name=&quot;VisibilityModifier&quot;&gt;
   &lt;property name=&quot;allowPublicImmutableFields&quot; value=&quot;true&quot;/&gt;
-  &lt;property name=&quot;immutableClassCanonicalNames&quot; value=&quot;
-  com.google.common.collect.ImmutableSet&quot;/&gt;
+  &lt;property name=&quot;immutableClassCanonicalNames&quot;
+        value=&quot;com.google.common.collect.ImmutableSet, java.lang.String&quot;/&gt;
 &lt;/module&gt;
         </source>
         <p>
           Example of allowed public immutable fields:
         </p>
         <source>
-public class ImmutableClass
-{
-  public final ImmutableSet&lt;String&gt; includes; // No warning
-  public final ImmutableSet&lt;String&gt; excludes; // No warning
-  public final java.lang.String notes; // Warning here because
-                                       //'java.lang.String' wasn't specified as allowed class
-  public final int someValue; // No warning
-
-  public ImmutableClass(Collection&lt;String&gt; includes, Collection&lt;String&gt; excludes,
-               String notes, int someValue)
-  {
-    this.includes = ImmutableSet.copyOf(includes);
-    this.excludes = ImmutableSet.copyOf(excludes);
-    this.value = value;
-    this.notes = notes;
-    this.someValue = someValue;
-  }
+public final class MyClass {
+ private int myPrivateField1;
+
+ int field1;               // violation, must have a visibility modifier
+
+ protected String field2;  // violation, protected visibility is not allowed
+
+ public int field3 = 42;  // violation, not static final, immutable,
+                          // nor matching the pattern
+ public long serialVersionUID = 1L;
```

Comment:
The violation for `field3` can be vertically aligned, please do it at all the places.

Reply:
This will make the violation message too long, we need to consider mobile users too.

Reply:
```suggestion

 int field1;               // violation, must have a visibility modifier

 protected String field2;  // violation, protected visibility is not allowed

 public int field3 = 42;   // violation, not static final, immutable,
                           // nor matching the pattern
 public long serialVersionUID = 1L;
```

Reply:
See diff above at https://github.com/checkstyle/checkstyle/pull/12897#discussion_r1150578726